include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2477,569 +2496,520 @@ namespace internal
+                                      2 * comp * dofs_per_face][v],
+                                temp1[reorientate(v, i) + dofs_per_face +
+                                      2 * comp * dofs_per_face][v],
+                                global_vector_ptr
+                                  [comp * static_dofs_per_component +
+                                   index_array_hermite[v][2 * i] +
+                                   dof_info.component_dof_indices_offset
+                                     [active_fe_index]
+                                     [first_selected_component] +
+                                   indices[v]],
+                                global_vector_ptr
+                                  [comp * static_dofs_per_component +
+                                   index_array_hermite[v][2 * i + 1] +
+                                   dof_info.component_dof_indices_offset
+                                     [active_fe_index]
+                                     [first_selected_component] +
+                                   indices[v]],
+                                grad_weight[v]);
                         }
                     }
+                }
               else
                 {
-                  if (integrate == false)
-                    for (unsigned int i = 0; i < n_components * dofs_per_face;
-                         ++i)
-                      temp1[i] = VectorizedArrayType();
-
-                  for (unsigned int v = 0; v < n_filled_lanes; ++v)
+                  for (unsigned int i = 0; i < dofs_per_face; ++i)
                     for (unsigned int comp = 0; comp < n_components; ++comp)
-                      for (unsigned int i = 0; i < dofs_per_face; ++i)
-                        function_3b(
-                          temp1[reorientate(n_face_orientations == 1 ? 0 : v,
-                                            i) +
-                                2 * comp * dofs_per_face][v],
-                          global_vector_ptr
-                            [indices[v] +
-                             (comp * static_dofs_per_component +
-                              index_array_nodal
-                                [n_face_orientations == 1 ? 0 : v][i]) *
-                               strides[v]]);
-                }
-            }
-        }
-
-      // case 4: contiguous indices without interleaving
-      else if (((do_gradients == false &&
-                 data.data.front().nodal_at_cell_boundaries == true) ||
-                (data.element_type ==
-                   MatrixFreeFunctions::tensor_symmetric_hermite &&
-                 fe_degree > 1)) &&
-               dof_info.index_storage_variants[dof_access_index][cell] ==
-                 MatrixFreeFunctions::DoFInfo::IndexStorageVariants::
-                   contiguous &&
-               dof_info.n_vectorization_lanes_filled[dof_access_index][cell] ==
-                 VectorizedArrayType::size())
```

Comment:
I have removed this check and have implemented the version if not all lanes are filled here.